src/main/java/com/telenordigital/decisionflow/DecisionFlow.java

```diff
@@ -0,0 +1,641 @@
+package com.telenordigital.decisionflow;
+
+import com.telenordigital.decisionflow.Decision.OnAttributesCallback;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.Callback;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.ElementDescriptor;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.ElementType;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+public class DecisionFlow<C, P> implements DecisionMachine<C, P> {
+
+    private InitialNode initialNode = null;
+    private final Map<String, AbstractNode> nodeMap = new HashMap<>();
+
+    private DecisionFlow(final DecisionFlowDescriber describer) {
+        load(describer);
+    }
+
+    public static <C, P> DecisionFlow<C, P> getInstance(final DecisionFlowDescriber describer) {
```

Comment:
If a constructor does significantly more than merely storing its arguments, I generally prefer a static factory method to it. Nonetheless, you still have a point here, bearing in mind that I wanted to discourage inheritance of this class, so I've made the following changes:

1. I've changed the declaration of getInstance() method to return DecisionMachine interface rather than DecisionFlow, which is more correct (obviously, I had made a kind of typo here without having realised that :)
2. To discourage inheritance even further, I've made class DecisionFlow final.